src/operation/find_and_modify.rs

```diff
@@ -21,89 +19,86 @@ use crate::{
         Namespace,
     },
     error::{ErrorKind, Result},
-    operation::{append_options, remove_empty_write_concern, OperationWithDefaults, Retryability},
+    operation::{
+        append_options_to_raw_document,
+        find_and_modify::options::Modification,
+        remove_empty_write_concern,
+        OperationWithDefaults,
+        Retryability,
+    },
     options::WriteConcern,
 };
 
-pub(crate) struct FindAndModify<T = Document>
-where
-    T: DeserializeOwned,
-{
+pub(crate) struct FindAndModify<'a, R, T: DeserializeOwned> {
     ns: Namespace,
     query: Document,
+    modification: Modification<'a, R>,
+    human_readable_serialization: Option<bool>,
```

Comment:
Could this be an un-serialized field in `FindAndModifyOptions` rather than adding it as an explicit parameter?